pepper-apis/dss-core/src/main/java/org/broadinstitute/ddp/route/DsmTriggerOnDemandActivityRoute.java

```diff
@@ -101,6 +112,25 @@ public Object handle(Request request, Response response, TriggerActivityPayload
                 handle.attach(DataExportDao.class).queueDataSync(user.getId(), studyId);
                 log.info("Created on-demand activity instance {} for study guid {}, activity code {}, participant guid {}",
                         instanceDto.getGuid(), studyGuid, activityCode, participantGuid);
+
+                //create answer for this instance if lms/osteo2:SOMATIC_RESULTS
+                if ((studyGuid.equalsIgnoreCase("cmi-lms") || studyGuid.equalsIgnoreCase("CMI-OSTEO"))
```

Comment:
study and activity specific logic which is a negative but that minimizes core DSS platform changes including NO new schema changes.

Reply:
Why do we need the "study" portion of the conditional?  If we decide to add the RESULT_FILE_ACTIVITY_ID to other studies, this code should work, right?

Reply:
I debated about it and tried to be extra cautious. I will remove study check. None of existing studies have this activity any way

Reply:
removed study specific check and added unit tests